src/UI/Art/SpriteArtist/SpriteLibrary.java

```diff
@@ -288,17 +278,103 @@ public static Sprite getMoveCursorArrowSprite()
 
     return moveCursorArrowSprite;
   }
+  
+  /**
+   * Sets up the available palettes for use.
+   * Reads in the first 6 colors on each row.
+   * Bottom colors are buildings, top are units.
+   * The key color is the last color on the top row.
+   */
+  private static void initResources()
+  {
+    if (null == mapUnitColorPalettes )
+    {
+      buildingColorPalettes = new HashMap<Color, ColorPalette>();
+      mapUnitColorPalettes = new HashMap<Color, ColorPalette>();
+      colorNames = new HashMap<Color, String>();
+      factionNames = new ArrayList<String>();
+
+      // Create a mapping of game colors to the fine-tuned colors that will be used for map sprites.
+      buildingColorPalettes.put(Color.PINK, new ColorPalette(pinkMapBuildingColors));
+      buildingColorPalettes.put(Color.CYAN, new ColorPalette(cyanMapBuildingColors));
+      buildingColorPalettes.put(Color.ORANGE, new ColorPalette(orangeMapBuildingColors));
+      buildingColorPalettes.put(PURPLE, new ColorPalette(purpleMapBuildingColors));
+
+      mapUnitColorPalettes.put(Color.PINK, new ColorPalette(pinkMapUnitColors));
+      mapUnitColorPalettes.put(Color.CYAN, new ColorPalette(cyanMapUnitColors));
+      mapUnitColorPalettes.put(Color.ORANGE, new ColorPalette(orangeMapUnitColors));
+      mapUnitColorPalettes.put(PURPLE, new ColorPalette(purpleMapUnitColors));
+
+      // Throw some color names in there for the defaults
+      // toString() is not user-friendly
+      colorNames.put(Color.PINK, "salmon");
+      colorNames.put(Color.CYAN, "cyan");
+      colorNames.put(Color.ORANGE, "citrus");
+      colorNames.put(PURPLE, "sparking");
 
-  private static ColorPalette getBuildingColors(Color colorKey)
+      // We want to be able to use the normal units, as well as any others
+      factionNames.add(DEFAULT_SPRITE_KEY);
+
+      final File folder = new File("res/unit/faction");
+
+      if (folder.canRead())
+      {
+        for( final File fileEntry : folder.listFiles() )
+        {
+          // If it's a file, we assume it's a palette
+          if( !fileEntry.isDirectory() )
+          {
+            String colorName = fileEntry.getAbsolutePath();
+            if( colorName.contains(".png") )
+            {
+              BufferedImage bi = loadSpriteSheetFile(colorName);
+              // Grab the last color on the first row as our "banner" color
+              Color key = new Color(bi.getRGB(bi.getWidth() - 1, 0));
+              Color[] cUnits = new Color[defaultMapColors.length];
+              Color[] cStructs = new Color[defaultMapColors.length];
+              for( int i = 0; i < defaultMapColors.length; i++ )
```

Comment:
Oh. Probably the "banner" color is last just to make this indexing more natural, huh.